cron_scripts/password_expiration_notification.pl

```diff
@@ -0,0 +1,279 @@
+        my $i    = 0;
+        foreach my $acc ( $searched_users->dataof("$path/account") ) {
+            $i++;
+            my $msom = $searched_users->match("$path/[$i]")
+              or die("MATCH '$path/[$i]' failed\n");
+
+            my $users_attr = getSoapResponseData($acc);
+            if ( defined $users_attr->{'zimbraPasswordModifiedTime'} ) {
+                my $zimbraPasswordModifiedTime =
+                  $users_attr->{'zimbraPasswordModifiedTime'};
+                $zimbraPasswordModifiedTime =
+                  substr $zimbraPasswordModifiedTime, 0, 12;
+                my $pass_expiration_year_days_users =
+                  Time::Piece->strptime( $zimbraPasswordModifiedTime,
+                    "%Y%m%d%H%M%S" );
+                my $expire_in_days =
+                  $pass_expiration_year_days -
+                  $pass_expiration_year_days_users->yday;
+                $users_attr->{'expire_in_days'} = $expire_in_days;
+            }
+            push @data, $users_attr;
+        }
+    }
+    else {
+        warn(   "Error: SearchDirectoryRequest "
+              . $soap_api->Error
+              . " via $config->{MAILSTOREHOST} \n" );
+    }
+    if ( scalar @data != 0 ) {
+        my $pm = Parallel::ForkManager->new( $opts{max_proc} );
+      ACCOUNTS:
+        foreach my $acc (@data) {
+            $pm->start and next ACCOUNTS;
+            my $mail           = $acc->{mail};
+            my $mobile         = $acc->{mobile};
+            my $displayName    = $acc->{displayName} || 'User';
+            my $expire_in_days = $acc->{'expire_in_days'};
+            if ($mail) {
+                my $mail_content = $config->{mail_content};
+                $mail_content =~ s/{displayName}/$displayName/g;
+                $mail_content =~ s/{numDays}/$expire_in_days/g;
+                my $mail_data = {
+                    'to' => $mail,
+                    'subject' =>
+"Your Mail access password expire in $acc->{'expire_in_days'} days",
+                    'body' => $mail_content
+                };
+                if ( my $sendmail =
+                    eval { $soap_api->sendmessage($mail_data) } )
+                {
+                    warn("DEBUG: Mail sent to $mail\n") if ($Debug);
+                }
+                else {
+                    warn( "Error: for $mail on sending mail. Message: "
+                          . ZCS::CustomAPI->Error );
+                }
+            }
+            if ($mobile) {
+                my $sms_content = $config->{sms_content};
+                $sms_content =~ s/{displayName}/$displayName/g;
+                $sms_content =~ s/{numDays}/$expire_in_days/g;
+                my $sms_url =
+"$config->{SMSHOST}?username=$config->{SMSUSER}&pin=$config->{SMSPIN}&mnumber=$mobile&message=$sms_content";
+
+#https://smsgw.sms.gov.in/failsafe/HttpLink?username=zimbra.sms&pin=moorxu81&message=hifromnic&mnumber=919922429908&signature=NICSMS
```

Comment:
Do we want to leave this example in?  Remove/redact/make generic the hostname and the pin?